bundle/src/main/java/com/meltmedia/dropwizard/etcd/cluster/ClusterService.java

```diff
@@ -171,35 +172,60 @@ public void stop() {
     }
 
     private ClusterAssignmentService assignments;
+    private ClusterAssignmentTracker tracker;
+    private ProcessorStateTracker processorStateTracker;
     private ClusterProcessor<C> processor;
-    private MappedEtcdDirectory<ClusterProcess> directory;
+    private EtcdDirectory directory;
+    private MappedEtcdDirectory<ClusterProcess> processDirectory;
+    private MappedEtcdDirectory<ProcessorNode> processorDirectory;
+    private ClusterNode thisNode;
 
-    public ProcessService(ClusterNode thisNode, ClusterStateTracker stateTracker,
-      ScheduledExecutorService executor, MappedEtcdDirectory<ClusterProcess> directory,
+    public ProcessService(ClusterNode thisNode, ClusterStateTracker stateTracker, ScheduledExecutorService executor, EtcdDirectory directory,
       Function<C, ClusterProcessLifecycle> lifecycleFactory, TypeReference<C> type,
       ObjectMapper mapper, MetricRegistry registry) {
+      this.thisNode = thisNode;
+      String dirName = directory.getName().replace('.', '_');
+      Function<String, String> metricName = name->MetricRegistry.name(ClusterAssignmentService.class, dirName, name);
+      this.processDirectory = directory.newDirectory("/processes", new TypeReference<ClusterProcess>(){});
+      this.processorDirectory = directory.newDirectory("/processors", new TypeReference<ProcessorNode>(){});
+      this.tracker = new ClusterAssignmentTracker(thisNode, processDirectory, registry, metricName);
+      this.processorStateTracker = ProcessorStateTracker.builder().withDirectory(processorDirectory).withThisNode(thisNode).build();
       this.assignments =
-        ClusterAssignmentService.builder().withExecutor(executor).withProcessDir(directory)
-          .withThisNode(thisNode).withClusterState(stateTracker).withMetricRegistry(registry).build();
+        ClusterAssignmentService.builder()
+          .withExecutor(executor).withProcessDir(processDirectory)
+          .withProcessorState(processorStateTracker::getState)
+          .withAssignmentState(tracker::getState)
+          .withThisNode(thisNode)
+          .withClusterState(stateTracker).withMetricRegistry(registry).withMetricName(metricName).build();
       this.processor =
         ClusterProcessor.<C> builder().withLifecycleFactory(lifecycleFactory)
-          .withDirectory(directory).withType(type).withNodeId(thisNode.getId()).withMapper(mapper)
+          .withDirectory(processDirectory).withType(type).withNodeId(thisNode.getId()).withMapper(mapper)
           .build();
       this.directory = directory;
     }
 
     public void start() {
+      processorStateTracker.start();
       processor.start();
+      tracker.start();
       assignments.start();
+      processorStateTracker.afterStart();
```

Comment:
Needing to call 2 methods like this, in sequence seems like trouble.  Perhaps it should be more like:

```
processorStateTracker.start(()->{
  processor.start();
  tracker.start();
  assignments.start();
});
```